src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -818,6 +818,7 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                 } else if (callback instanceof CredentialCallback) {
                     final CredentialCallback credentialCallback = (CredentialCallback) callback;
 
+                    // TODO: should consider realm from RealmCallback
```

Comment:
Looking for way how to obtain PasswordCredential with specified realm. But to obtain TwoWayPassword and convert to digest password of given realm would be maybe also appropriate...
In both cases I think it should be in standalone PR, so only TODO keeping here.